pkg/vm/memory/relocatable.go

```diff
@@ -168,3 +168,36 @@ func (m MaybeRelocatable) AddMaybeRelocatable(other MaybeRelocatable) (MaybeRelo
 		return MaybeRelocatable{}, errors.New("RelocatableAdd")
 	}
 }
+
+func (m MaybeRelocatable) SubMaybeRelocatable(other MaybeRelocatable) (MaybeRelocatable, error) {
+	m_felt, m_is_felt := m.GetFelt()
+	other_felt, other_is_felt := other.GetFelt()
+
+	m_rel, m_is_rel := m.GetRelocatable()
+	other_rel, other_is_rel := other.GetRelocatable()
+
+	if m_is_felt && other_is_felt {
+		return *NewMaybeRelocatableFelt(m_felt.Sub(other_felt)), nil
+	}
+
+	if m_is_rel && other_is_rel {
+		if m_rel.SegmentIndex == other_rel.SegmentIndex {
+			felt := lambdaworks.FeltFromUint64(uint64(m_rel.Offset - other_rel.Offset))
+			return *NewMaybeRelocatableFelt(felt), nil
+		} else {
+			return MaybeRelocatable{}, errors.New("RelocatableSubDiffIndex")
```

Comment:
I think we should change the error message to something like `Can't subtract two relocatable values with different segment indexes`